fvtest/compilertriltest/LongAndAsRotateTest.cpp

```diff
@@ -0,0 +1,494 @@
+                                                            0xFFFFFFFFFFFF0000,
+                                                            0xFFFF000000000000                     
+                                                        };
+
+// this vector is for tests where contiguous zeros only appear once in the binary value.
+// ex: 11100001111, 1000001, 0000000
+std::vector<uint64_t> lconstInput_onesAroundZeros = { 0x0, 0x8000000000000001, 0xE000000000000003, 0xFFFFFFF00FFFFFFF, 0xFFFFFFFEFFFFFFFF, 0xFFFFFFFE7FFFFFFF, 0xF0FFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFF0F, 0XFFFF000000000001, 0x8000FFFFFFFFFFFF };
+
+// These input values cannot be handled by a RISBG instruction
+std::vector<uint64_t> lconstInput_invalidPatterns = {0x4000000000000001, 0xF0F0F0F0F0F0F0F0, 0x8000000000000002};
+
+template <typename VarType, typename ConstType>
+using LongAndAsRotateParamType = std::tuple<VarType, ConstType, ConstType (*) (VarType, ConstType)>;
+
+template <typename VarType, typename ConstType>
+class LongAndAsRotateTest : public ::testing::TestWithParam<LongAndAsRotateParamType<VarType, ConstType>>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      const char *options = "-Xjit:disableTreeSimplification,disableAsyncCompilation,acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck";
+      auto initSuccess = initializeJitWithOptions(const_cast<char*>(options));
+
+      ASSERT_TRUE(initSuccess) << "Failed to initialize JIT.";   
+      }
+
+   static void TearDownTestCase()
+      {
+      shutdownJit();
+      }
+   };
+
+template <typename VarType, typename ConstType>
+struct LongAndAsRotateParamStruct
+   {
+   VarType variableValue;  //not known at compile time
+   ConstType constantValue;  //known at compile time
+   ConstType (*oracle)(VarType, ConstType);
+   };
+
+template <typename VarType, typename ConstType>
+LongAndAsRotateParamStruct<VarType, ConstType> to_struct(LongAndAsRotateParamType<VarType, ConstType> param)
+   {
+   LongAndAsRotateParamStruct<VarType, ConstType> s;
+
+   s.variableValue = std::get<0>(param);
+   s.constantValue = std::get<1>(param);
+   s.oracle = std::get<2>(param);
+
+   return s;   
+   }
+
+template <typename VarType, typename ConstType>
+static ConstType longAndAsRotateOracle(VarType a, ConstType b)
+   {
+   return ((ConstType)a & b);
+   }
+
+class i2lLongAndAsRotateTest : public LongAndAsRotateTest<int32_t, uint64_t> {};
+
+TEST_P(i2lLongAndAsRotateTest, SimpleTest)
+   {
+   auto param = to_struct(GetParam());
+
+   char inputTrees[512] = {0};
+   std::snprintf(inputTrees, sizeof(inputTrees),
+         "(method return=Int64 args=[Int32]  "
+         " (block                            "
+         "  (lreturn                         "
+         "  (land                            " 
+         "      (i2l (iload parm=0))         "
+         "      (lconst %s)))))              ",
+         std::to_string(param.constantValue).c_str()
```

Comment:
Sorry but this isn't going to work on our build compiler. 

I believe `std::to_string` only got a constructor overload for `int` starting `C++11`. 

A possible workaround would be to do the following :

```
...

(lconst %lld) ))))) 

...
param.constantValue

```

Please do the same for the rest of the cases where you are using ` std::to_string(param.constantValue).c_str()` 
is used.